compiler/util/misc.cpp

```diff
@@ -171,6 +172,20 @@ const char* cleanFilename(const BaseAST* ast) {
   return retval;
 }
 
+static const char* cleanCompilerFilename(const char* name) {
```

Comment:
Could we use the `stripDirectories` function in `compiler/util/files.cpp` for this rather than introduce a new function?

Reply:
I still think it makes sense to add the new function (since it indicates what we are doing at a high level) but I've adjusted it to call the existing `stripdirectories`.